src/librustc/middle/const_eval.rs

```diff
@@ -416,6 +424,12 @@ pub enum ErrKind {
     ExpectedConstTuple,
     ExpectedConstStruct,
     TupleIndexOutOfBounds,
+    IndexedNonVec,
+    IndexNotNatural,
```

Comment:
Nit: I feel like I vaguely remember what "natural numbers" are from grade-school, but wouldn't "positive" be easier?